### DISCUSSION TITLE
Gain and Amplitude

### DISCUSSION THREAD

**Discussion by Bumblebee001** [GrandOrgue/GoOdf#45] [Ideas]
Do we really need these to go up by 0.000001 step at a time? 

Wouldn't it make more sense to go in whole numbers rather than 10 e-6 fractions?

**Comment by larspalo** [comment]
I've tried, but unfortunately without resorting to splitting the wxSpinCtrlDouble into a separate wxTextCtrl and a wxSpinCtrl (which is how it's done in GrandOrgue code, but I don't want to do that as I've mentioned before because of the large amount of extra code needed for that) I've not (yet) been able to find a solution that both allows decimal points and increments by whole numbers. If I set the increment to whole numbers it's no longer possible to enter a fraction even in the text box (at least under wxGTK).

**Reply by Bumblebee001** [reply]
Why should we want decimal points at all?

**Reply by larspalo** [reply]
The only reason is because it's possible to have fractional values in GrandOrgue for them. Why should GOODF put limits on values where GO does not?

**Reply by Bumblebee001** [reply]
Those fractions are meaningless in my humble opinion. The changes are far too small (imperceptable) for any practical relevance. Unless I am missing something.

**Reply by larspalo** [reply]
You may have your opinion, but as long as GO supports float values for them GOODF must support reading and writing them too. Practical? I've certainly never felt the need for decimals with AmplitudeLevel when voicing pipes - but I sometimes use values like 0.001 for MinVelocityVolume with noise effects that I want to be velocity sensitive, and that has actually the same linear scale as AmplitudeLevel.

A change of 1 in AmplitudeLevel corresponds to a change of about 0.09 dB (Gain). For Gain it makes more sense with decimals, but the important point is that GOODF should conform to GO specs.

**Reply by Bumblebee001** [reply]
Perhaps we may need to get Oleg to consider reducing the decimal placings at least to just 2 rather than 6 in GO! What do you think?
